src/restconfig/src/main/java/org/geoserver/rest/security/UrlCheckController.java

```diff
@@ -0,0 +1,206 @@
+        }
+
+        return wrapObject(check, AbstractURLCheck.class);
+    }
+
+    @PostMapping(
+            consumes = {
+                MediaType.TEXT_XML_VALUE,
+                MediaType.APPLICATION_XML_VALUE,
+                MediaTypeExtensions.TEXT_JSON_VALUE,
+                MediaType.APPLICATION_JSON_VALUE
+            })
+    @ResponseStatus(HttpStatus.CREATED)
+    public ResponseEntity<String> urlCheckPost(
+            @RequestBody AbstractURLCheck urlCheck, UriComponentsBuilder builder) {
+        try {
+
+            if (urlCheckDao.getCheckByName(urlCheck.getName()) != null) {
+                throw new RestException(
+                        "URL check '" + urlCheck.getName() + "' already exists",
+                        HttpStatus.CONFLICT);
+            }
+
+            verifyCheckName(urlCheck);
+            verifyCheckConfiguration(urlCheck);
+
+            /* enforce URL check enabled status by default */
+            if (urlCheck.isEnabled() == null) {
+                urlCheck.setEnabled(true);
+            }
+            urlCheckDao.save(urlCheck);
+
+            String name = urlCheck.getName();
+
+            LOGGER.log(Level.FINEST, "Added urlCheck {0}", name);
+
+            return new ResponseEntity<>(
+                    name, composeResponseHeaders(builder, name), HttpStatus.CREATED);
+
+        } catch (IOException ex) {
+            throw new RestException(
+                    "Error occurred in creating a new URL check",
+                    HttpStatus.INTERNAL_SERVER_ERROR,
+                    ex);
+        }
+    }
+
+    private static void verifyCheckName(AbstractURLCheck urlCheck) {
+        String checkName = urlCheck.getName();
+        if (checkName == null || checkName.isEmpty()) {
+            throw new RestException("The URL check name is required", HttpStatus.BAD_REQUEST);
+        }
+    }
+
+    private HttpHeaders composeResponseHeaders(UriComponentsBuilder builder, String name) {
+        HttpHeaders headers = new HttpHeaders();
+        UriComponents uriComponents = composeUriComponents(name, builder);
+        headers.setLocation(uriComponents.toUri());
+        headers.setContentType(MediaType.TEXT_PLAIN);
+        return headers;
+    }
+
+    private UriComponents composeUriComponents(String name, UriComponentsBuilder builder) {
+        return builder.path("/urlchecks/{id}").buildAndExpand(name);
+    }
+
+    @PutMapping(
+            value = "/{urlCheckName}",
+            consumes = {
+                MediaType.TEXT_XML_VALUE,
+                MediaType.APPLICATION_XML_VALUE,
+                MediaTypeExtensions.TEXT_JSON_VALUE,
+                MediaType.APPLICATION_JSON_VALUE
+            })
+    @ResponseStatus(HttpStatus.OK)
+    public void urlCheckPut(
+            @RequestBody AbstractURLCheck providedCheck, @PathVariable String urlCheckName) {
+        try {
+
+            AbstractURLCheck urlCheck = urlCheckDao.getCheckByName(urlCheckName);
```

Comment:
The API is generic, don't use setters but a reflection API to update the fields instead (e.g., OWSUtils). 
For example LayerController delegate the update to CatalogBuilder, which in turn ends up doing the following:

``
OwsUtils.copy(update, original, clazz);
``

Reply:
Fixed. Thanks.